tests/device_tests/ethereum/test_definitions.py

```diff
@@ -0,0 +1,231 @@
+    # Any chain id is allowed as long as the SLIP44 stays the same
+    params = DEFAULT_TX_PARAMS.copy()
+    params.update(chain_id=222222)
+    ethereum.sign_tx(client, **params)
+
+
+def test_slip44_disallowed(client: Client) -> None:
+    # SLIP44 is not allowed without a valid network definition
+    params = DEFAULT_TX_PARAMS.copy()
+    params.update(n=parse_path("m/44h/66666h/0h/0/0"))
+    with pytest.raises(TrezorFailure, match="Forbidden key path"):
+        ethereum.sign_tx(client, **params)
+
+
+def test_slip44_external(client: Client) -> None:
+    # to use a non-default SLIP44, a valid network definition must be provided
+    network = common.encode_network(chain_id=66666, slip44=66666)
+    params = DEFAULT_TX_PARAMS.copy()
+    params.update(n=parse_path("m/44h/66666h/0h/0/0"), chain_id=66666)
+    ethereum.sign_tx(client, **params, definitions=common.make_defs(network, None))
+
+
+def test_slip44_external_disallowed(client: Client) -> None:
+    # network definition does not allow a different SLIP44
+    network = common.encode_network(chain_id=66666, slip44=66666)
+    params = DEFAULT_TX_PARAMS.copy()
+    params.update(n=parse_path("m/44h/55555h/0h/0/0"), chain_id=66666)
+    with pytest.raises(TrezorFailure, match="Forbidden key path"):
+        ethereum.sign_tx(client, **params, definitions=common.make_defs(network, None))
+
+
+def test_chain_id_mismatch(client: Client) -> None:
+    # network definition for a different chain id will be rejected
+    network = common.encode_network(chain_id=66666, slip44=60)
+    params = DEFAULT_TX_PARAMS.copy()
+    params.update(chain_id=55555)
+    with pytest.raises(TrezorFailure, match="Network definition mismatch"):
+        ethereum.sign_tx(client, **params, definitions=common.make_defs(network, None))
+
+
+def test_definition_does_not_override_builtin(client: Client) -> None:
+    # The builtin definition for Ethereum (SLIP44 60, chain_id 1) will be used
+    # even if a valid definition with a different SLIP44 is provided
+    network = common.encode_network(chain_id=1, slip44=66666)
+    params = DEFAULT_TX_PARAMS.copy()
+    params.update(n=parse_path("m/44h/66666h/0h/0/0"), chain_id=1)
+    with pytest.raises(TrezorFailure, match="Forbidden key path"):
+        ethereum.sign_tx(client, **params, definitions=common.make_defs(network, None))
+
+    # TODO: test that the builtin definition will not show different symbol
+
+
+# TODO: figure out how to test acceptance of a token definition
+# all tokens are currently accepted, we would need to check the screenshots
+
+
+def test_builtin_token(client: Client) -> None:
+    # The builtin definition for USDT (ERC20) will be used even if not provided
+    params = DEFAULT_ERC20_PARAMS.copy()
+    params.update(to=ERC20_BUILTIN_TOKEN)
+    ethereum.sign_tx(client, **params)
+    # TODO check that USDT symbol is shown
+
+
+# TODO: test_builtin_not_overriden (builtin definition is used even if a custom one is provided)
```

Comment:
Done in `test_definition_does_not_override_builtin`?

Reply:
no, this is for tokens. added `token_not_overriden`
0ec31fe38676161e3a78b204e5ffddfba167fcd9